addons/obiaway/Obiaway.lua

```diff
@@ -137,9 +164,146 @@ function inTable(tbl, item)
     return false
 end
 
-function get_all_elements()
+-- converts name of a bag (str) to an id number. returns 0 (inventory) when no argument passed.
+function inv_str_to_id(str)
+    if not str then return 0 end
+    if str == 'inventory' then
+        return 0
+    elseif str == 'safe' then
+        return 1
+    elseif str == 'storage' then
+        return 2
+    elseif str == 'locker' then
+        return 3
+    elseif str == 'satchel' then
+        return 5
+    elseif str == 'sack' then
+        return 6
+    elseif str == 'case' then
+        return 7
+    elseif str == 'wardrobe' then
+        return 8
+    end
+end
+
+-- Function which counts how many slots remain in a bag. if no location is passed, checks inventory.
+function free_space(location)
+    local id = inv_str_to_id(location)
+
+    local inv = windower.ffxi.get_items(id)
```

Comment:
While this works, you should use `windower.ffxi.get_bag_info` here, as it is much more efficient. `windower.ffxi.get_items` will push all items in that particular bag onto the Lua stack and pack them into tables with all their fields (ID, count, slot, extdata, etc.). `windower.ffxi.get_bag_info` is precisely there to prevent these expensive calls if all you want is the bag meta-data and not the contents.